utils/node_local_test.py

```diff
@@ -470,11 +470,47 @@ def fetch_containers(self):
 class DaosCont():
     """Class to store data about daos containers"""
 
-    # pylint: disable=too-few-public-methods
-
-    def __init__(self, cont_uuid, label):
+    def __init__(self, cont_uuid, label, pool=None):
         self.uuid = cont_uuid
         self.label = label
+        self.pool = pool
+
+    # pylint: disable-next=invalid-name
+    def id(self):
+        """Return the pool ID (label if set; UUID otherwise)"""
+        if self.label:
+            return self.label
+        return self.uuid
+
+    def set_attrs(self, attrs):
+        """Set container attributes.
+
+        Args:
+            attrs (dict): Dictionary of attributes to set.
+        """
+        if not isinstance(self.pool, DaosPool):
+            raise NLTestFail('No pool passed when creating container')
+        for key, value in attrs.items():
+            rc = run_daos_cmd(self.pool.conf, ['container', 'set-attr', self.pool.id(), self.uuid,
+                                               f'{key}:{value}'],
+                              show_stdout=True)
+            print(rc)
+            assert rc.returncode == 0, rc
+
+    def destroy(self, valgrind=True, log_check=True):
+        """Destroy the container
+
+        Args:
+            valgrind (bool, optional): Run the command under valgrind. Defaults to True.
+            log_check (bool, optional): Run log analysis. Defaults to True.
+
+        Raises:
+            NLTestFail: If Pool was not provided when object created.
+        """
+        if not isinstance(self.pool, DaosPool):
+            raise NLTestFail('No pool passed when creating container')
```

Comment:
This is intentional.  If a DaosPool object is passed when DaosCont is created then the DaosCont object can save extra data in the object which means destroy can work.  If a string is passed in for pool then it can't and this method will not work.  Future work might be to only accept a DaosPool when creating which would simplify this.

Reply:
Ok, I did this.  There are a couple of cases where DaosCont is created through a path so picks up the pool from there and therefore there is no DaosPool object to provide, but those tests don't actually destroy the container afterwards so they work just fine without this check.